pkg/test/testPackages.proj

```diff
@@ -0,0 +1,77 @@
+<?xml version="1.0" encoding="utf-8"?>
+<Project ToolsVersion="14.0" DefaultTargets="Build" xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
+  <Import Project="$([MSBuild]::GetDirectoryNameOfFileAbove($(MSBuildThisFileDirectory), dir.props))\dir.props" />
+
+  <ItemGroup>
+    <PackageReports Condition="'$(TestPackage)' == ''" Include="$(PackageReportDir)*.json" />
+    <PackageReports Condition="'$(TestPackage)' != ''" Include="$(PackageReportDir)$(TestPackage).json" />
+
+    <!-- restrict restore sources to the current build and publicly shipped packages -->
+    <RestoreSource Include="$(PackageOutputPath)" />
+    <RestoreSource Include="https:%2F%2Fapi.nuget.org/v3/index.json" />
+
+    <!-- needed for SQLClient's SNI packages -->
+    <RestoreSource Include="https:%2F%2Fdotnet.myget.org/F/dotnet-core/api/v3/index.json" />
```

Comment:
I'm not especially worried, but I do think we need some restore validation. (EG: no downgrades, exact match)